R/mwrite.R

```diff
@@ -0,0 +1,415 @@
+#' can be useful for (1) breaking connection to NONMEM modeling outputs that 
+#' are imported by `$NMXML` or `$NMEXT` and (2) saving model updates (e.g., 
+#' an updated parameter list). Models can be read back using [mread_yaml()] or 
+#' converted to mrgsolve cpp format with [yaml_to_cpp()].
+#' 
+#' @param x a model object. 
+#' @param file output file name; if non-character (e.g., `NULL`), no output 
+#' will be written to file. 
+#' @param digits precision to use when writing outputs. 
+#' 
+#' @details
+#' Parameters and omega and sigma matrices that were imported via `$NMXML`
+#' or `$NMEXT` will be written into the yaml file and the NONMEM import blocks 
+#' will be dropped. This allows the user to load a model based on a NONMEM run 
+#' without having a connection to that output (e.g., `root.xml` or `root.ext`). 
+#' Given that the connection to the NONMEM modeling outputs is broken when 
+#' writing to yaml, any update to the NONMEM run will only be propagated to 
+#' the yaml file when `mwrite_yaml()` is run again. 
+#' 
+#' The yaml file does not currently have the ability to track 
+#' other external dependencies, such as user-defined header files or other 
+#' code that might be sourced in by the user when the model is loaded via 
+#' [mread()]. NONMEM xml and ext files imported by `$NMXML` or `$NMEXT` are 
+#' the _only_ external dependencies that are accounted for in the yaml 
+#' transport file. 
+#' 
+#' @return 
+#' A list containing data that was written out to the yaml file, with added 
+#' item `file`, is returned invisibly. 
+#' 
+#' @examples
+#' mod <- house()
+#' 
+#' temp1 <- tempfile(fileext = ".yaml")
+#' 
+#' x <- mwrite_yaml(mod, temp1)
+#' 
+#' readLines(temp1)
+#' 
+#' @seealso [mread_yaml()], [yaml_to_cpp()]
+#' 
+#' @md
+#' @export
+mwrite_yaml <- function(x, file, digits = 8) {
+  require_yaml()
+  l <- mwrite_model_to_list(x)
+  l$format <- "yaml"
+  out <- yaml::as.yaml(l, precision = digits)
+  if(is.character(file)) {
+    writeLines(con = file, out)  
+  }
+  l$format <- "list"
+  l$file <- file
+  invisible(l)
+}
+
+#' Write a model to native mrgsolve format
+#' 
+#' Model code is written to a file in native mrgsolve format. This 
+#' can be useful for (1) breaking connection to NONMEM modeling outputs that 
+#' are imported by `$NMXML` or `$NMEXT` and (2) saving model updates (e.g., 
+#' an updated parameter list). Models can be read back using [mread()].
+#' 
+#' @inheritParams mwrite
+#' @inheritParams yaml_to_cpp
+#' 
+#' @details
+#' See important details in [mwrite_yaml()].
+#' 
+#' @return 
+#' A list containing data that was written out to the cpp file, with added 
+#' item `file`, is returned invisibly. 
+#' 
+#' @examples
+#' temp <- tempfile(fileext = ".mod")
+#' 
+#' mod <- modlib("pk1", compile = FALSE)
+#' 
+#' x <- mwrite_cpp(mod, file = temp)
+#' 
+#' mod <- mread(x$file, compile = FALSE)
+#' 
+#' mod
+#' 
+#' @seealso [mwrite_yaml()], [yaml_to_cpp()]
+#' 
+#' @md
+#' @rdname mwrite
```

Comment:
Drop this `rdname` and just produce `mwrite_cpp.Rd` (consistent with `mwrite_yaml` mapping to `mwrite_yaml.Rd`)?

Reply:
Done.